code-challenges.test.js

```diff
@@ -14,38 +16,125 @@
 
 // a) Create a test with an expect statement using the variable provided.
 
-const people = [
-  { name: "ford prefect", occupation: "a hitchhiker" },
-  { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "a radio employee" }
-]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
-
+// 1a) Pseudo Code:
+// Copy and paste test code template into assignment (make sure nothing is inside initial template)
+// Read problem statement and understand inputs/outputs (inputs need to be directly inserted into test code)
+// Create relevant info for describe, it, expect, and toEqual methods
+// Test code output to ensure it is failing red (should get a reference error saying function is not defined)
+
+describe("capitalizerPartTwo", () => {
+  it("a function which takes in an array of objects, which returns an array of strings which capitalizes the name of each person and combines it with their occupation", () => {
+    expect(
+      capitalizerPartTwo([
+        { name: "ford prefect", occupation: "a hitchhiker" },
+        { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+        { name: "arthur dent", occupation: "a radio employee" },
+      ])
+    ).toEqual([
+      "Ford Prefect is a hitchhiker.",
+      "Zaphod Beeblebrox is president of the galaxy.",
+      "Arthur Dent is a radio employee.",
+    ]);
+  });
+});
 
 // b) Create the function that makes the test pass.
 
+// 1b) Pseudo Code:
+// Create the function and give it a parameter (array)
+// Use a form of iteration to iterate through the array (return the result so you don't have to create a variable)
+// Iteration logic:
+//    Use a method to capitalize each name (or potentially break the name apart into two names and .toUpperCase the index of the first letter, and then recombine them)
+//    Use string interpolation to create the sentence for each entry (make sure you refer to each key properly)
+// Run test code to ensure it is passing green
+
+const capitalizerPartTwo = (array) => {
+  return array.map((value) => {
+    let uncapitalizedName = value.name.split(" ");
+    let capitalizedName = uncapitalizedName.map((value) => {
+      return value.slice(0, 1).toUpperCase() + value.slice(1, value.length + 1);
+    });
+    capitalizedName = capitalizedName.join(" ");
+    return `${capitalizedName} is ${value.occupation}.`;
+  });
+};
+
+// Notes: I ran the initial test and got the expected error. While writing the code, I realized I had to do a little more logic for the capitalization part. I thought there was a method already made which could do just that. So instead I had to use slice to capitalize each starting character and then use concatenations. The logic is a bit bulky so definitely needs a refactor. Salvador told me I could use .select instead, so I will try to explore using that if time permits.
```

Comment:
This looks great. Love the descriptive variable names. The only refactor is cleaning up `slice`. It can take two arguments but if you only pass one the second will automatically be the length. `return value.slice(0, 1).toUpperCase() + value.slice(1)`